python/graphstorm/config/argument.py

```diff
@@ -748,19 +752,72 @@ def eval_fanout(self):
             # By default use -1 as full neighbor
             return [-1] * self.num_layers
 
+    @property
+    def textual_data_path(self):
+        """ distillation textual data path
+        """
+        if hasattr(self, "_textual_data_path"):
+            return self._textual_data_path
+        return None
+
+    @property
+    def lm_name(self):
+        """ HuggingFace Transformer-based model name for distillation
+        """
+        if hasattr(self, "_lm_name"):
+            return self._lm_name
+        return None
+
+    @property
+    def pre_trained_name(self):
+        """ Name of pre-trained HuggingFace model weights for distillation
+        """
+        if hasattr(self, "_pre_trained_name"):
+            return self._pre_trained_name
+        return None
+
+    @property
+    def max_global_step(self):
+        """ Maximum global size for distillation.
+        """
+        # only needed by distillation
+        if "distill" not in self.configuration["gsf"]:
+            return None
+        if hasattr(self, "_max_global_step"):
+            return self._max_global_step
+        else:
+            # default max global steps
+            return 10000
+
+    @property
+    def max_seq_len(self):
+        """ Maximum sequence length for distillation.
+        """
+        # only needed by distillation
+        if "distill" not in self.configuration["gsf"]:
+            return None
+        if hasattr(self, "_max_seq_len"):
+            return self._max_seq_len
+        else:
+            # default max global steps
+            return 1024
+
     @property
     def hidden_size(self):
         """ Hidden embedding size
         """
         # pylint: disable=no-member
-        assert hasattr(self, "_hidden_size"), \
-            "hidden_size must be provided when pretrain a embedding layer, " \
-            "or train a GNN model"
-        assert isinstance(self._hidden_size, int), \
-            "Hidden embedding size must be an integer"
-        assert self._hidden_size > 0, \
-            "Hidden embedding size must be larger than 0"
-        return self._hidden_size
+        if "distill" not in self.configuration["gsf"]:
```

Comment:
Does the distill logic call this function? If not, you do not need to change this function.
You only need to update the ```verify_arguments``` function. Furthermore, ```verify_arguments``` already skips checking hidden_size when encoder_type is "lm".

Reply:
encoder_type is under "gsf -> gnn" section in the yaml config, but the config for distillation doesn't have this section.

Reply:
ditto